contracts/voting/dao-voting-cw721-staked/src/contract.rs

```diff
@@ -343,22 +337,57 @@ pub fn execute_claim_nfts(
     deps: DepsMut,
     env: Env,
     info: MessageInfo,
+    token_ids: Option<Vec<String>>,
 ) -> Result<Response, ContractError> {
-    let nfts = NFT_CLAIMS.claim_nfts(deps.storage, &info.sender, &env.block)?;
-    if nfts.is_empty() {
-        return Err(ContractError::NothingToClaim {});
-    }
+    let token_ids = match token_ids {
+        // attempt to claim all legacy NFTs
+        None => {
+            // claim all legacy NFTs
```

Comment:
We should probably have a check here that ensures we don't have a claim in the new system.

Reply:
ah, the goal of this fix is to be backwards compatible so DAOs can seamlessly migrate. both the legacy and non-legacy claim systems will exist simultaneously, until all legacy claims expire. at that point, we could remove all the legacy logic, but for now we need to support claiming both.

Reply:
This logic I described should actually be in unstake, but it's only using the new system there now. Looks good here then